app/javascript/app/service-provider-form.js

```diff
@@ -1,52 +1,71 @@
-$(function () {
-  if (!$('.service-provider-form').length) {
-    return;
-  }
+const showElement = (element) => element.classList.remove('display-none');
+const hideElement = (element) => element.classList.add('display-none');
 
+function ialOptionSetup() {
   // Selectors
-  const idProtocol = $('input[name="service_provider[identity_protocol]"]');
-  const ialLevel = $('#service_provider_ial');
-  const samlFields = $('.saml-fields');
-  const oidcFields = $('.oidc-fields');
-  const ialAttributesCheckboxes = $('.ial-attr-wrapper');
-  const fileContainer = $('#certificate-container');
-  const logoInput = $('.input-file');
-  const pemInputMessage = $('.js-pem-input-error-message');
-  const pemInput = $('.js-pem-input');
-  const redirectURI = $('#add-redirect-uri-field');
-  const failureToProofURL = $('.service_provider_failure_to_proof_url');
-  const returnToSpUrl = document.querySelector('#service_provider_return_to_sp_url');
-
+  const ialLevel = document.getElementById('service_provider_ial');
+  const ialAttributesCheckboxes = document.querySelectorAll('.ial-attr-wrapper');
+  const failureToProofURL = document.querySelector('.service_provider_failure_to_proof_url');
   const ial1Attributes = ['email', 'all_emails', 'x509_subject', 'x509_presented', 'verified_at'];
 
   // Functions
   const toggleIAL1Options = () => {
-    ialAttributesCheckboxes.each((idx, attr) => {
-      const element = $(attr).find('input');
-
-      if (!ial1Attributes.includes(element.val())) {
-        $(attr).hide();
-        element.prop('checked', false);
+    ialAttributesCheckboxes.forEach((checkboxWrapper) => {
+      const checkboxInput = checkboxWrapper.querySelector('input');
+      if (ial1Attributes.includes(checkboxInput.value)) {
+        hideElement(checkboxWrapper);
+        checkboxInput.checked = false;
       }
     });
   };
 
   const toggleIAL2Options = () => {
-    ialAttributesCheckboxes.each((idx, attr) => $(attr).show());
+    ialAttributesCheckboxes.forEach((checkboxWrapper) => {
+      showElement(checkboxWrapper);
+    });
+  };
+
+  const toggleIALOptions = (ial) => {
+    switch (ial) {
+      case '1':
+        hideElement(failureToProofURL);
+        toggleIAL1Options();
+        break;
+      case '2':
+        showElement(failureToProofURL);
+        toggleIAL2Options();
+        break;
+      default:
+        showElement(failureToProofURL);
+        toggleIAL2Options();
+    }
   };
 
+  // Page initialization
+  toggleIALOptions(ialLevel.value);
```

Comment:
Apparently we load this script on the "Show" page for an SP, triggering an error at this line. I assume it should only be loaded on the edit form itself?

```
Uncaught TypeError: Cannot read properties of null (reading 'value')
```